edge-util/src/Microsoft.Azure.Devices.Edge.Util/azureLogAnalytics/AzureLogAnalytics.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Security.Cryptography;
+    using System.Text;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    /* Sample code from:
+    /* https://github.com/veyalla/MetricsCollector/blob/master/modules/MetricsCollector/AzureLogAnalytics.cs
+    /* https://dejanstojanovic.net/aspnet/2018/february/send-data-to-azure-log-analytics-from-c-code/
+    */
+
+    public sealed class AzureLogAnalytics
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<AzureLogAnalytics>();
+        static AzureLogAnalytics instance = null;
+        public string WorkspaceId { get; }
+        public string SharedKey { get; }
+        public string ApiVersion { get; }
+        AzureLogAnalytics(string workspaceId, string sharedKey)
+        {
+            const string apiVersion = "2016-04-01";
+
+            Preconditions.CheckNotNull(workspaceId, "Log Analytic workspace ID cannot be empty.");
+            Preconditions.CheckNotNull(sharedKey, "Log Analytic shared key cannot be empty.");
+
+            this.WorkspaceId = workspaceId;
+            this.SharedKey = sharedKey;
+            this.ApiVersion = apiVersion;
+        }
+
+        public static AzureLogAnalytics GetInstance()
+        {
+            Preconditions.CheckNotNull(instance);
+            return instance;
+        }
+
+        public static AzureLogAnalytics InitInstance(string workspaceId, string sharedKey)
```

Comment:
This method looks weird as the caller doesn't know if it creates a new instance with given parameter or use existing one.  let discuss tomorrow.

Reply:
Per discussion in-person, the `workdspaceId`, and `sharedKey` are moved into `PostAsync()`